src/constraint_system/composer.rs

```diff
@@ -84,6 +85,28 @@ impl StandardComposer {
     pub fn circuit_size(&self) -> usize {
         self.n
     }
+
+    /// Constructs a dense vector of the Public Inputs from the positions and the
+    /// sparse vector that contains the values.
+    pub(crate) fn construct_dense_pi_vec(&self) -> Vec<BlsScalar> {
+        let mut pi = vec![BlsScalar::zero(); self.n];
+        self.public_inputs_sparse_store
+            .keys()
+            .zip(self.public_inputs_sparse_store.values())
+            .for_each(|(pos, value)| {
+                pi[*pos] = *value;
+            });
```

Comment:
You're zipping yourself. :) It make sense when you have the keys and the values on two different objects, but here the object is just one. You should be able just to iterate it to get the entries (a tuple `(key, value)`).